### PR TITLE
Add missing options in model scaffolding

### PR DESCRIPTION
**Pull request by Samuell1**
Adding missing options can speed up development a lot. First i wanted to add it as "--all" option but stub files doesnt support if statements.

### CODE REVIEW

File: src/Scaffold/Console/model/model.stub [hunk 1]
```diff
@@ -22,6 +22,28 @@ class {{studly_name}} extends Model
      */
     protected $fillable = [];
 
+    /**
+     * @var array Rules for fields
+     */
+    protected $rules = [];
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: Rules is a public property not protected and it requires the validation trait to be included.

File: src/Scaffold/Console/model/model.stub [hunk 2]
```diff
@@ -22,6 +22,28 @@ class {{studly_name}} extends Model
...
+     * @var array Json fields
+     */
+    protected $jsonable = [];
+
+    /**
+     * @var array Date fields
+     */
+    protected $date = [
+        'created_at',
+    ];
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: The dates casting is done with `dates` not `date` and exists on the base model covering `created_at` and `updated_at` so it should not be encouraged to modify it unless actually necessary by adding more dates columns. If an argument can be made for including it in the stub then it should default to the defaults in the base model

Comment (Samuell1) [reply src/Scaffold/Console/model/model.stub]: If i make it empty array it will still use predefined dates in Model?

Comment (LukeTowers) [reply src/Scaffold/Console/model/model.stub]: No, which means you'd have to either not include it or include it with the same defaults (i.e. just add `updated_at`)

File: src/Scaffold/Console/model/model.stub [hunk 3]
```diff
@@ -22,6 +22,39 @@ class {{studly_name}} extends Model
      */
     protected $fillable = [];
 
+    /**
+     * @var array Rules for fields
+     */
+    public $rules = [];
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: You need to add the validation trait to make this one take effect

File: src/Scaffold/Console/model/model.stub [hunk 4]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
...
+     */
+    protected $appends = [];
+
+    /**
+     * @var array Hide attributes
+     */
+    protected $hidden = [];
+
+    /**
+     * @var array Date fields
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: Comment should be `Attributes to be cast to Carbon instances`

File: src/Scaffold/Console/model/model.stub [hunk 5]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
...
+     */
+    protected $jsonable = [];
+
+    /**
+     * @var array Append attributes
+     */
+    protected $appends = [];
+
+    /**
+     * @var array Hide attributes
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: Comment should be `Attributes to be removed from the API representation of the model (ex. toArray())`

File: src/Scaffold/Console/model/model.stub [hunk 6]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
...
+     */
+    protected $casts = [];
+
+    /**
+     * @var array Json fields
+     */
+    protected $jsonable = [];
+
+    /**
+     * @var array Append attributes
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: `Attributes to be appended to the API representation of the model (ex. toArray())`

File: src/Scaffold/Console/model/model.stub [hunk 7]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
...
+     */
+    public $rules = [];
+
+    /**
+     * @var array Cast fields
+     */
+    protected $casts = [];
+
+    /**
+     * @var array Json fields
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: `Attributes to be cast to JSON`

File: src/Scaffold/Console/model/model.stub [hunk 8]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
      */
     protected $fillable = [];
 
+    /**
+     * @var array Rules for fields
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: `Validation rules for attributes`

File: src/Scaffold/Console/model/model.stub [hunk 9]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
...
+     */
+    public $rules = [];
+
+    /**
+     * @var array Cast fields
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]: `The attributes that should be cast to native types.`

**Comment by w20k** [comment]
@LukeTowers - LGTM

Screenshot:
Screenshot 2019-07-29 at 20 48 46

File: src/Scaffold/Console/model/model.stub [hunk 10]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
      */
     protected $fillable = [];
 
+    /**
+     * @var array Validation rules for attributes
+     */
+    public $rules = [];
+
+    /**
+     * @var array The attributes that should be cast to native types
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]:
```suggestion
     * @var array Attributes that to be cast to native types
```

File: src/Scaffold/Console/model/model.stub [hunk 11]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
...
+     */
+    protected $appends = [];
+
+    /**
+     * @var array Attributes to be removed from the API representation of the model (ex. toArray())
+     */
+    protected $hidden = [];
+
+    /**
+     * @var array Attributes to be cast to Carbon instances
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]:
```suggestion
     * @var array Attributes to be cast to Argon (Carbon) instances
```

File: src/Scaffold/Console/model/model.stub [hunk 12]
```diff
@@ -22,6 +24,39 @@ class {{studly_name}} extends Model
      */
     protected $fillable = [];
 
+    /**
+     * @var array Validation rules for attributes
+     */
+    public $rules = [];
+
+    /**
+     * @var array Attributes that to be cast to native types
```
Comment (LukeTowers) [inline src/Scaffold/Console/model/model.stub]:
```suggestion
     * @var array Attributes to be cast to native types
```

**Comment by Samuell1** [comment]
@LukeTowers pinging you back

**Pull request merged**